file.cc

```diff
@@ -236,30 +130,30 @@ char * Class::gets( char * s, int size, bool stripNl )
   return result;
 }
 
-std::string Class::gets( bool stripNl ) 
+std::string Class::gets( bool stripNl )
 {
-  char buf[ 1024 ];
+  auto line = QString( f.readLine() );
 
-  if ( !gets( buf, sizeof( buf ), stripNl ) )
+  if ( line.isEmpty() )
     throw exReadError();
 
-  return std::string( buf );
+  return stripNl ? line.trimmed().toStdString() : line.toStdString();
 }
```

Comment:
the original code seems only trimmed right space.

Reply:
Yes, u r right :sweat_smile: